File: backend/src/main/kotlin/ch/yvu/teststore/statistics/TestStatistics.kt
package ch.yvu.teststore.statistics

import ch.yvu.teststore.common.Model
import com.datastax.driver.mapping.annotations.PartitionKey
import com.datastax.driver.mapping.annotations.Table
import java.util.*

@Table(name = "test_statistics")
data class TestStatistics(
        @PartitionKey(0) var testSuite: UUID?,
        @PartitionKey(1) var testName: String?,
        var numPassed: Int?,
        var numFailed: Int?): Model {
    constructor(): this(null, null, null, null)
}

File: web/www/app/statistics/test-statistics.ts
export class TestStatistics {
    testSuite: string;
    testName: string;
    numPassed: number;
    numFailed: number;
    durationSum: number;
    durationNum: number;

    getTotalResults(): number {
        return this.numPassed + this.numFailed;
    }

    getPassRate(): number {
        let totalResults = this.getTotalResults();
        if(totalResults == 0) return 0;
        return this.numPassed / totalResults;
    }

    getAverageDuration(): number {
        if(this.durationNum == 0) return 0;
        return this.durationSum / this.durationNum;
    }
}